### ISSUE TITLE
Paddle2.2 linux 编译报错

### ISSUE THREAD

**Issue by Deo0330** [PaddlePaddle/Paddle#37179]
1）PaddlePaddle版本：2.2
    2）系统环境：aistudio 项目环境 16G GPU Python3.7

1）docker编译
参照官方编译步骤（linux ubuntu Docker GPU编译）
重新按照步骤编译了也还是跳出一样的错误，是gcc 7.5.0 不支持编译吗
报错如下，不知道该如何修改，望回复
[  0%] copy_if_different /home/aistudio/Paddle/build/paddle/fluid/inference/api/paddle_inference_pass.h
[  0%] copy_if_different /home/aistudio/Paddle/build/paddle/fluid/pybind/pybind.h
[  0%] copy_if_different /home/aistudio/Paddle/build/paddle/fluid/operators/jit/kernels.h
[  0%] Built target extern_pocketfft
[  1%] Built target extern_xbyak
[  1%] Built target extern_cryptopp
[  1%] Built target extern_gloo
[  1%] Built target extern_pybind
[  1%] Built target extern_cub
[  1%] Built target extern_mkldnn
[  1%] Built target extern_utf8proc
[  1%] Built target copy_pybind_command
[  2%] Built target extern_dlpack
[  2%] Built target extern_mklml
[  2%] Built target extern_threadpool
[  2%] Built target download_externalError
[  2%] Built target copy_kernels_command
[  2%] Built target extern_gflags
[  3%] Built target extern_lapack
[  3%] Built target copy_paddle_inference_pass_command
[  3%] Built target extern_zlib
[  3%] Built target extern_eigen3
[  3%] Built target extern_boost
[  3%] Built target extern_warpctc
[  3%] Built target profiler_py_proto_init
[  3%] Built target framework_py_proto_init
[  3%] Built target fleet_proto_init
[  4%] Built target extern_protobuf
[  4%] Built target extern_glog
[  4%] Built target mkldnn_cmd
Copy generated python proto into directory paddle/fluid/proto/profiler.
[  4%] Built target distributed_strategy_py_proto
[  4%] Built target timer
[  4%] Built target string_array
[  4%] Built target trainer_py_proto
[  4%] Built target pass_desc_py_proto
[  4%] Built target table_printer
[  4%] Built target activation_functions
[  4%] Built target profiler_py_proto
[  5%] Built target zero_copy_tensor_dummy
[  5%] Built target paddle_pass_builder
[  5%] Built target shape_range_info_proto
[  5%] Built target error_codes_proto
[  5%] Built target external_error_proto
Copy generated python proto into directory paddle/fluid/proto.
[  5%] Built target framework_proto
[  6%] Built target monitor
[  6%] Built target heter_service_proto
[  6%] Built target mkldnn
[  6%] Built target cudnn_workspace_helper
[  6%] Built target denormal
[  6%] Built target version
[  6%] Built target data_feed_proto
[  6%] Built target flags
[  6%] Built target framework_py_proto
[  6%] Built target profiler_proto
[  6%] Built target errors
[  6%] Built target op_def_proto
[  6%] Built target op_version_proto
[  6%] Built target pass_desc_proto
[  6%] Built target string_helper
[  6%] Built target pretty_log
[  6%] Built target stringpiece
[  6%] Built target imperative_flag
[  6%] Built target prune
[  6%] Built target trainer_desc_proto
[  7%] Built target enforce
[  7%] Built target op_def_api
[  7%] Built target imperative_profiler
[  7%] Built target extern_xxhash
[  7%] Built target eigen_function
[  7%] Built target op_version_registry
[  7%] Built target cpu_info
[  7%] Built target ddim
[  7%] Built target dynamic_loader
[  7%] Built target threadpool
[  7%] Built target shell
[  7%] Built target place
[  7%] Built target stream_callback_manager
[  7%] Built target cuda_profiler
[  7%] Built target attribute
[  7%] Built target fs
[  7%] Built target workqueue
[  7%] Built target data_loader
[  7%] Built target benchmark
[  7%] Built target third_party
[  7%] Built target cuda_stream
[  7%] Built target dynload_warpctc
[  8%] Built target dynload_mklml
[  8%] Built target paddle_crypto
[  8%] Built target dynload_lapack
[  8%] Built target cblas
[  8%] Built target allocator
[  8%] Built target generator
[  8%] Built target memory_block
[  8%] Built target dynload_cuda
[  8%] Built target op_proto_maker
[  8%] Built target cpu_helper
[  8%] Built target thread_local_allocator
[  8%] Built target aligned_allocator
[  8%] Built target cpu_allocator
[ 10%] Built target pinned_allocator
[ 10%] Built target locked_allocator
[ 10%] Built target best_fit_allocator
[ 10%] Built target retry_allocator
[ 10%] Built target mmap_allocator
[ 10%] Built target buffered_allocator
[ 10%] Built target op_call_stack
[ 10%] Built target jit_kernel_base
[ 10%] Built target gpu_info
[ 10%] Built target device_tracer
[ 10%] Built target auto_growth_best_fit_allocator
[ 10%] Built target cuda_device_guard
[ 10%] Built target jit_kernel_mix
[ 10%] Built target jit_kernel_mkl
[ 10%] Built target jit_kernel_refer
[ 10%] Built target cuda_resource_pool
[ 11%] Built target system_allocator
[ 11%] Built target device_memory_aligment
[ 11%] Built target profiler
[ 11%] Built target jit_kernel_intrinsic
[ 11%] Built target cuda_allocator
[ 11%] Built target jit_kernel_jitcode
[ 11%] Built target buddy_allocator
[ 12%] Built target jit_kernel_helper
[ 12%] Built target naive_best_fit_allocator
[ 12%] Built target allocator_strategy
[ 12%] Built target allocator_facade
[ 12%] Built target malloc
[ 12%] Built target device_context
[ 12%] Built target prelu
[ 12%] Built target data_type
[ 13%] Built target sampler
[ 14%] Built target device_code
[ 14%] Built target bert_encoder_functor
[ 14%] Building CUDA object paddle/fluid/operators/math/CMakeFiles/lstm_compute.dir/lstm_compute.cu.o
[ 14%] Built target shape_inference
[ 14%] Built target matrix_bit_code
[ 14%] Built target op_kernel_type
[ 14%] Built target heter_wrapper
[ 14%] Built target sequence_padding
[ 14%] Built target blas
[ 14%] Built target unpooling
[ 14%] Built target memcpy
[ 14%] Built target collective_helper
[ 14%] Built target vol2col
[ 14%] Built target im2col
[ 14%] Built target sequence_scale
[ 14%] Built target no_need_buffer_vars_inference
[ 14%] Built target segment_pooling
[ 14%] Built target concat_and_split
[ 14%] Built target lapack_function
[ 14%] Built target mkldnn_axpy_handler
[ 14%] Built target matrix_inverse
[ 14%] Built target depthwise_conv
[ 14%] Built target sample_prob
[ 14%] Built target memory
[ 14%] Built target cross_entropy
[ 14%] Built target op_info
[ 14%] Built target maxouting
[ 14%] Built target pooling
[ 14%] Built target sequence2batch
[ 14%] Built target cos_sim_functor
[ 14%] Built target unused_var_check
[ 15%] Built target fc
[ 15%] Built target math_function
[ 15%] Built target matrix_solve
[ 15%] Built target mask_util
[ 15%] Building CUDA object paddle/fluid/framework/CMakeFiles/tensor.dir/tensor_util.cu.o
[ 15%] Built target garbage_collector
[ 15%] Building CUDA object paddle/fluid/operators/math/CMakeFiles/softmax.dir/softmax.cu.o
[ 15%] Built target tree2col
[ 15%] Built target context_project
[ 15%] Building CUDA object paddle/fluid/operators/math/CMakeFiles/gru_compute.dir/gru_compute.cu.o
[ 15%] Built target beam_search
[ 15%] Built target sequence_pooling
/home/aistudio/Paddle/paddle/fluid/framework/tensor_util.cu(1): error: this declaration has no storage class or type specifier

/home/aistudio/Paddle/paddle/fluid/framework/tensor_util.cu(1): error: expected a ";"

2 errors detected in the compilation of "/tmp/tmpxft_00000c61_00000000-6_tensor_util.cpp1.ii".
paddle/fluid/framework/CMakeFiles/tensor.dir/build.make:75: recipe for target 'paddle/fluid/framework/CMakeFiles/tensor.dir/tensor_util.cu.o' failed
make[2]: *** [paddle/fluid/framework/CMakeFiles/tensor.dir/tensor_util.cu.o] Error 1
CMakeFiles/Makefile2:8355: recipe for target 'paddle/fluid/framework/CMakeFiles/tensor.dir/all' failed
make[1]: *** [paddle/fluid/framework/CMakeFiles/tensor.dir/all] Error 2
make[1]: *** 正在等待未完成的任务....
/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(1761): error: identifier "__builtin_ia32_sqrtsd_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(1770): error: identifier "__builtin_ia32_sqrtss_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(2728): error: identifier "__builtin_ia32_scalefsd_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(2737): error: identifier "__builtin_ia32_scalefss_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(11265): error: identifier "__builtin_ia32_scalefsd_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(11274): error: identifier "__builtin_ia32_scalefss_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1163): error: identifier "__builtin_ia32_reducesd" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1171): error: identifier "__builtin_ia32_reducess" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1179): error: identifier "__builtin_ia32_rangesd128_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1189): error: identifier "__builtin_ia32_rangess128_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1198): error: identifier "__builtin_ia32_rangesd128_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1207): error: identifier "__builtin_ia32_rangess128_round" is undefined

12 errors detected in the compilation of "/tmp/tmpxft_00000bcc_00000000-6_lstm_compute.cpp1.ii".
paddle/fluid/operators/math/CMakeFiles/lstm_compute.dir/build.make:75: recipe for target 'paddle/fluid/operators/math/CMakeFiles/lstm_compute.dir/lstm_compute.cu.o' failed
make[2]: *** [paddle/fluid/operators/math/CMakeFiles/lstm_compute.dir/lstm_compute.cu.o] Error 1
CMakeFiles/Makefile2:85085: recipe for target 'paddle/fluid/operators/math/CMakeFiles/lstm_compute.dir/all' failed
make[1]: *** [paddle/fluid/operators/math/CMakeFiles/lstm_compute.dir/all] Error 2
/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(1761): error: identifier "__builtin_ia32_sqrtsd_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(1770): error: identifier "__builtin_ia32_sqrtss_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(2728): error: identifier "__builtin_ia32_scalefsd_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(2737): error: identifier "__builtin_ia32_scalefss_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(11265): error: identifier "__builtin_ia32_scalefsd_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(11274): error: identifier "__builtin_ia32_scalefss_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1163): error: identifier "__builtin_ia32_reducesd" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1171): error: identifier "__builtin_ia32_reducess" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1179): error: identifier "__builtin_ia32_rangesd128_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1189): error: identifier "__builtin_ia32_rangess128_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1198): error: identifier "__builtin_ia32_rangesd128_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1207): error: identifier "__builtin_ia32_rangess128_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(1761): error: identifier "__builtin_ia32_sqrtsd_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(1770): error: identifier "__builtin_ia32_sqrtss_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(2728): error: identifier "__builtin_ia32_scalefsd_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(2737): error: identifier "__builtin_ia32_scalefss_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(11265): error: identifier "__builtin_ia32_scalefsd_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512fintrin.h(11274): error: identifier "__builtin_ia32_scalefss_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1163): error: identifier "__builtin_ia32_reducesd" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1171): error: identifier "__builtin_ia32_reducess" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1179): error: identifier "__builtin_ia32_rangesd128_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1189): error: identifier "__builtin_ia32_rangess128_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1198): error: identifier "__builtin_ia32_rangesd128_round" is undefined

/usr/lib/gcc/x86_64-linux-gnu/7/include/avx512dqintrin.h(1207): error: identifier "__builtin_ia32_rangess128_round" is undefined

12 errors detected in the compilation of "/tmp/tmpxft_00000c96_00000000-6_gru_compute.cpp1.ii".
paddle/fluid/operators/math/CMakeFiles/gru_compute.dir/build.make:75: recipe for target 'paddle/fluid/operators/math/CMakeFiles/gru_compute.dir/gru_compute.cu.o' failed
make[2]: *** [paddle/fluid/operators/math/CMakeFiles/gru_compute.dir/gru_compute.cu.o] Error 1
CMakeFiles/Makefile2:84904: recipe for target 'paddle/fluid/operators/math/CMakeFiles/gru_compute.dir/all' failed
make[1]: *** [paddle/fluid/operators/math/CMakeFiles/gru_compute.dir/all] Error 2
12 errors detected in the compilation of "/tmp/tmpxft_00000c86_00000000-6_softmax.cpp1.ii".
paddle/fluid/operators/math/CMakeFiles/softmax.dir/build.make:75: recipe for target 'paddle/fluid/operators/math/CMakeFiles/softmax.dir/softmax.cu.o' failed
make[2]: *** [paddle/fluid/operators/math/CMakeFiles/softmax.dir/softmax.cu.o] Error 1
CMakeFiles/Makefile2:85184: recipe for target 'paddle/fluid/operators/math/CMakeFiles/softmax.dir/all' failed
make[1]: *** [paddle/fluid/operators/math/CMakeFiles/softmax.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by ronny1996** [comment]
您好，推荐使用paddle官方的docker镜像编译
